core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -501,21 +589,26 @@ int linkEntry(int ei, K key) {
     /**
      * write value in place
      **/
-    void writeValue(int hi, V value) {
-        assert hi >= 0;
-        ByteBuffer byteBuffer = memoryManager.allocate(valueSerializer.calculateSize(value));
-        // just allocated bytebuffer is ensured to have position 0
-        valueSerializer.serialize(value, byteBuffer.slice());
-        handles[hi].setValue(byteBuffer);
+    long writeValue(V value) {
+        // the length of the given value plus its header
+        int valueLength = valueSerializer.calculateSize(value) + ValueUtils.VALUE_HEADER_SIZE;
+        Slice slice = memoryManager.allocateSlice(valueLength);
```

Comment:
Please add a comment that allocateSlice does not creates a new ByteBuffer object as part of the Slice, the ByteBuffer object is created later when provided to value serializer. If it indeed works as I think...

Reply:
I do not understand what you mean in this comment by "... does not create a new ByteBuffer ... the ByteBuffer is created later ...".